x-pack/plugins/security/public/management/roles/roles_grid/roles_grid_page.tsx

```diff
@@ -259,16 +258,64 @@ export class RolesGridPage extends Component<Props, State> {
     ] as Array<EuiBasicTableColumn<Role>>;
   };
 
-  private getVisibleRoles = () => {
-    const { roles, filter } = this.state;
+  private getVisibleRoles = (roles: Role[], filter: string, includeReservedRoles: boolean) => {
+    return roles.filter(role => {
+      const normalized = `${role.name}`.toLowerCase();
+      const normalizedQuery = filter.toLowerCase();
+      return (
+        normalized.indexOf(normalizedQuery) !== -1 &&
+        (includeReservedRoles || !isRoleReserved(role))
+      );
+    });
+  };
+
+  private onIncludeReservedRolesChange = (e: EuiSwitchEvent) => {
```

Comment:
Can you add a test to ensure that changing this switch hides the reserved roles?